include/networkit/graph/DHBGraph.hpp

```diff
@@ -0,0 +1,1991 @@
+
+    /**
+     * Increase the weight of an edge. If the edge does not exist,
+     * it will be inserted.
+     *
+     * @param[in]	u	endpoint of edge
+     * @param[in]	v	endpoint of edge
+     * @param[in]	weight	edge weight
+     */
+    void increaseWeight(node u, node v, edgeweight ew);
+
+    /* SUMS */
+
+    /**
+     * Returns the sum of all edge weights.
+     * @return The sum of all edge weights.
+     */
+    edgeweight totalEdgeWeight() const noexcept;
+
+    /**
+     * Get an iterable range over the nodes of the graph.
+     *
+     * @return Iterator range over the nodes of the graph.
+     */
+    NodeRange nodeRange() const noexcept { return NodeRange(*this); }
+
+    /**
+     * Get an iterable range over the edges of the graph.
+     *
+     * @return Iterator range over the edges of the graph.
+     */
+    EdgeRange edgeRange() const noexcept { return EdgeRange(*this); }
+
+    /**
+     * Get an iterable range over the edges of the graph and their weights.
+     *
+     * @return Iterator range over the edges of the graph and their weights.
+     */
+    EdgeWeightRange edgeWeightRange() const noexcept { return EdgeWeightRange(*this); }
+
+    /**
+     * Get an iterable range over the neighbors of @a.
+     *
+     * @param u Node.
+     * @return Iterator range over the neighbors of @a u.
+     */
+    NeighborRange<false> neighborRange(node u) const {
+        assert(u < upperNodeIdBound());
+        return NeighborRange<false>(*this, u);
+    }
+
+    /**
+     * Get an iterable range over the neighbors of @a u including the edge
+     * weights.
+     *
+     * @param u Node.
+     * @return Iterator range over pairs of neighbors of @a u and corresponding
+     * edge weights.
+     */
+    NeighborWeightRange<false> weightNeighborRange(node u) const {
+        assert(isWeighted());
+        assert(u < upperNodeIdBound());
+        return NeighborWeightRange<false>(*this, u);
+    }
+
+    /**
+     * Default Graph class: Returns the index of node v in the array of outgoing edges of node u.
+     * DHB: Returns the index of node v in the neighbors of outgoing edges of node u.
+     * @param u Node
+     * @param v Node
+     * @return index of node v in the array of outgoing edges of node u.
+     * TODO: We might want to support concurrency for this function in the future.
+     */
+    index indexOfNeighbor(node u, node v) const;
+
+    /**
+     * Return the i-th (outgoing) neighbor of @a u.
+     *
+     * @param u Node.
+     * @param i index; should be in [0, degreeOut(u))
+     * @return @a i-th (outgoing) neighbor of @a u, or @c none if no such
+     * neighbor exists.
+     */
+    node getIthNeighbor(node u, index i) const;
+
+    /**
+     * Return the weight to the i-th (outgoing) neighbor of @a u.
+     *
+     * @param u Node.
+     * @param i index; should be in [0, degreeOut(u))
+     * @return @a edge weight to the i-th (outgoing) neighbor of @a u, or @c +inf if no such
```

Comment:
The function actually returns the default edge weight and not inf when the neighbor does not exist.